pdfminer/psparser.py

```diff
@@ -217,7 +218,7 @@ def seek(self, pos: int) -> None:
         self.buf = b''
         self.charpos = 0
         # reset the status for nexttoken()
-        self._parse1 = self._parse_main
+        self._parse1 = weakref.WeakMethod(self._parse_main)
```

Comment:
I'd be tempted to add a brief comment here explaining the use of a weak ref to avoid a cyclic reference.